pkg/resourceinterpreter/configurableinterpreter/configmanager/manager.go

```diff
@@ -0,0 +1,107 @@
+package configmanager
+
+import (
+	"fmt"
+	"sync/atomic"
+
+	"k8s.io/apimachinery/pkg/labels"
+	"k8s.io/apimachinery/pkg/runtime/schema"
+	utilruntime "k8s.io/apimachinery/pkg/util/runtime"
+	"k8s.io/client-go/tools/cache"
+	"k8s.io/klog/v2"
+
+	configv1alpha1 "github.com/karmada-io/karmada/pkg/apis/config/v1alpha1"
+	"github.com/karmada-io/karmada/pkg/util/fedinformer"
+	"github.com/karmada-io/karmada/pkg/util/fedinformer/genericmanager"
+	"github.com/karmada-io/karmada/pkg/util/helper"
+)
+
+var resourceInterpreterCustomizationsGVR = schema.GroupVersionResource{
+	Group:    configv1alpha1.GroupVersion.Group,
+	Version:  configv1alpha1.GroupVersion.Version,
+	Resource: "resourceinterpretercustomizations",
+}
+
+// ConfigManager can list custom resource interpreter.
+type ConfigManager interface {
+	LuaScriptAccessors() map[schema.GroupVersionKind]LuaScriptAccessor
+	HasSynced() bool
+}
+
+// interpreterConfigManager  collect the resource interpreter customization configuration.
+type interpreterConfigManager struct {
+	initialSynced *atomic.Value
+	lister        cache.GenericLister
+	configuration *atomic.Value
+}
+
+// LuaScriptAccessors return all configured resource by lua.
```

Comment:
```suggestion
// LuaScriptAccessors returns all cached configurations.
```
Given the configuration not only scripts, so the name should be changed, I'll think about it later.